src/main/kotlin/xyz/crossward/service/EntryService.kt

```diff
@@ -0,0 +1,35 @@
+package xyz.crossward.service
+
+import org.springframework.stereotype.Service
+import xyz.crossward.entities.Entry
+import xyz.crossward.exception.BadRequestException
+import xyz.crossward.repository.EntryRepository
+import java.time.LocalDate
+
+@Service
+class EntryService(
+    private val entryRepository: EntryRepository
+) {
+
+    /**
+     * Record a new entry into the database.
+     *
+     * If a date is not specified, then then the result of LocalDate.now() will be used
```

Comment:
This will take some thinking (and maybe we can think about it later as a seperate PR) but we should operate on the NYT schedule for the dates we store, to make it easy for storing crossword times for the current crossword day without worrying about specifying a date in the request.

Crosswords are available at the following times (in EST):
Monday: 6PM (Sunday) - 10PM (Monday)
Tuesday - Friday: 10PM (previous day) - 10PM
Saturday: 10PM (Friday) - 6PM
Sunday: 6PM (Saturday) - 6PM

So, if a user wants to record an entry and doesn't specify a date, we should have a function that looks at the current weekday and time returns the date for the current available crossword.

Reply:
Working with timezones has also always been a hassle for the bot. The NYT Crossword App always operates on Eastern Time (observing daylight savings), but I was never able to make my python code automatically adjust to Daylight Savings, the timezone libraries just dont seem to work with the bot's job scheduler, so I've had to manually adjust the UTC values myself twice a year. If we can figure out that whole can of worms too that would be much better too (can definitely be a different PR).

Reply:
We can potentially put all this logic in the frontend too.

Reply:
Another thought: I think it's fine to allow users to submit times for past crossword days (although maybe that messes with the concept of having a "Win" on your group's leaderboard) but we definitely shouldn't allow users to submit times for future crossword days